### PR TITLE
🏃[e2e] improve and cleanup e2e clusterctl

### PR DESCRIPTION
**Pull request by fabriziopandini**
**What this PR does / why we need it**:
This PR implements a set of small improvements and cleanup to the  e2e clusterctl package:
- removes `+build e2e` tag (not required in the framework)
- removes inconsistency around `LogPath`/`LogFolder` variable naming (now it is used consistently `LogFolder`)
- refactors `config.Validate` into two smaller functions (fixes a linter check)
- add few utility functions to the `config` type

**Which issue(s) this PR fixes**:
xref #2753
xref #2637
xref #2636

/assign @vincepri
/assign @sedefsavas

### CODE REVIEW

File: test/framework/clusterctl/clusterctl_config.go [hunk 1]
```diff
@@ -1,5 +1,3 @@
-// +build e2e
```
Comment (vincepri) [inline test/framework/clusterctl/clusterctl_config.go:1]: Don't we need this? @wfernandes has a different PR that adds this

**Comment by vincepri** [comment]
If we remove the build tag, `make test` will run also the tests in these files

File: test/framework/clusterctl/clusterctl_config.go (hunk 1 above)
Comment (fabriziopandini) [reply test/framework/clusterctl/clusterctl_config.go:1]: @vincepri @wfernandes 
The test framework is basically a library that CAPI is offering to external users, so IMO it should build (and run tests) like any other package

Comment (wfernandes) [reply test/framework/clusterctl/clusterctl_config.go:1]: FWIW I was adding a build tag to `docker/e2e/custom_management.go `

**Pull request merged**